[iOS & tvOS] Video Range Types

JPKribs opened the pull request:
### Summary

Resolves: https://github.com/jellyfin/Swiftfin/issues/1196

Adds SDR/HDR and Dolby Vision media attributes.

### Screenshots

<details>

<summary>HDR</summary>

<img width="383" alt="Screenshot 2025-03-10 at 14 41 35" src="https://github.com/user-attachments/assets/93b396f6-6cef-47a0-8129-84bf72cbef8f" />

</details>

<details>

<summary>SDR</summary>

<img width="825" alt="Screenshot 2025-03-10 at 14 40 54" src="https://github.com/user-attachments/assets/c0e34a11-a2c8-4c0b-a7c8-db9d4c921174" />

</details>

### Formatting Changes

<details>

<summary>Original</summary>

<img width="281" alt="Screenshot 2025-03-10 at 14 43 09" src="https://github.com/user-attachments/assets/0865bc1f-292a-43a7-bd69-71346cbf7bd9" />

</details>


<details>

<summary>Updated Formatting (iPhone SE)</summary>

This should be our smallest potential screen.

<img width="333" alt="Screenshot 2025-03-10 at 15 25 04" src="https://github.com/user-attachments/assets/2e42debf-a83e-4c2e-aa02-97761531c605" />

</details>

LePips reviewed (commented):
I changed the API a bit on the `AttributeBadge` and thank you for making that change

This single row is also becoming cluttered which was mentioned when we added the ratings. We may finally want to take a look at implementing something like [WrappingHStack](https://github.com/dkk/WrappingHStack) for these.

JPKribs commented on Swiftfin tvOS/Views/ItemView/Components/AttributeHStack.swift:
```diff
@@ -17,71 +17,72 @@ extension ItemView {
+                if viewModel.selectedMediaSource?.mediaStreams?.hasHDRVideo == true {
+                    AttributeBadge(style: .fill, title: "HDR")
+                } else {
+                    AttributeBadge(style: .fill, title: "SDR")
```
I agree! To me, it makes sense to do the same for 4K / HD. If it's 4K we don't need the HD badge as well. So something like:

```
                if viewModel.selectedMediaSource?.mediaStreams?.has4KVideo == true {
                    AttributeBadge(style: .fill, title: "4K")
                } else if viewModel.selectedMediaSource?.mediaStreams?.hasHDVideo == true {
                    AttributeBadge(style: .fill, title: "HD")
                }
```

Do you want me to make that change too or just HDR/SDR?

JPKribs commented:
> I changed the API a bit on the `AttributeBadge` and thank you for making that change
> 
> This single row is also becoming cluttered which was mentioned when we added the ratings. We may finally want to take a look at implementing something like [WrappingHStack](https://github.com/dkk/WrappingHStack) for these.

This is what this looks like using `WrappingHStack`. I had to rework the inside of `AttributeHStack` a lot for iOS since I needed to pass in all of the badges as independent views. If I grouped any, like all the video items, it would make this `WrappingHStack` not work as expected. Here is what this looks like now:

<img width="333" alt="Screenshot 2025-03-12 at 11 27 33" src="https://github.com/user-attachments/assets/d39eea57-e3e5-425a-90cc-417b9ffd50a6" />

---

Do we want this same thing on tvOS? I feel like tvOS we can get away with the simple `HStack` due to the increased screen real estate.

holow29 commented:
I don't want to add too much complexity here, but you might consider the following for HDR:
There are videos with HDR10+ (and therefore HDR10 fallback as well) and DV. In this case, it would be good to know it has both, though I think if someone sees HDR10+, HDR10 fallback can be safely assumed. Basically, in this case, I imagine two HDR badges: HDR10+, DV
The HDR badge seems redundant if it is showing another HDR-related badge like DV. If there is an HDR10 badge instead, then it would make sense to have both if the video supports both. Basically, I'm not sure the point of the HDR badge since we can be more specific.
Having badges for other HDR types like HLG might make sense.

I understand there is the constraint of the SDK as well, though as you noted some work with videoRangeTypes is being done AFAIK server-side, so maybe that will help give more options.

LePips reviewed (approved):
I don't mind adding HDR10 and HDR10+ in the future, we can just take the highest format.

The current work for `WrappingHStack` is good for now. When we drop iOS 15, which is soon, we should move to a custom `Layout` that exhibits the same behavior - CoPilot was able to make me a sufficient replacement.
